modules/storage/src/main/kotlin/edu/stanford/spezi/modules/storage/key/KeyValueStorageFactory.kt

```diff
@@ -36,23 +38,36 @@ internal class KeyValueStorageFactoryImpl @Inject constructor(
         fileName: String,
         type: KeyValueStorageType,
     ): Lazy<SharedPreferences> {
-        return lazy {
+        return lazy(LazyThreadSafetyMode.SYNCHRONIZED) {
             when (type) {
-                KeyValueStorageType.UNENCRYPTED -> context.getSharedPreferences(
-                    fileName,
-                    Context.MODE_PRIVATE
-                )
-
-                KeyValueStorageType.ENCRYPTED -> {
-                    EncryptedSharedPreferences.create(
-                        context,
-                        fileName,
-                        masterKey,
-                        EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
-                        EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
-                    )
+                KeyValueStorageType.UNENCRYPTED -> createUnencryptedStorage(fileName = fileName)
+
+                KeyValueStorageType.ENCRYPTED -> createEncryptedStorage(fileName = fileName) ?: run {
+                    logger.w { "First encrypted storage creation failed, deleting existing file and retrying..." }
+                    context.deleteSharedPreferences(fileName)
+                    createEncryptedStorage(fileName = fileName) ?: run {
+                        logger.w { "Second encrypted storage creation failed, returning a non encrypted file instead" }
+                        createUnencryptedStorage(fileName = fileName)
```

Comment:
Wondering if that would be unexpected behavior; it might make sense to fail at that point rather than storing it in an unencrypted way?

Reply:
We should never land at this point, as after deleting the previous file, we will be able to create the fresh encrypted file. I just wanted to avoid throwing at this point as it would result into a runtime crash. As I said I am pretty confident that we won't land here, and can adapt to fail here instead.

Reply:
Sounds good, if we should never get there crashing on purpose might be good; would allow us to collect crash reports and investigate why and how we could land here. Better than silently failing and storing the data in an unencrypted setup.